pyproject.toml

```diff
@@ -3,17 +3,17 @@
 # "Prerequisites" in User's Guide,
 # build jobs in `.github/workflows/*.yml`.
 requires = [
-    "setuptools >= 61.0.0",
+    "setuptools >=61.0.0",
     "wheel",
-    "numpy < 2",
+    "numpy >=2,<3",
```

Comment:
This is a point that probably we should discuss.
I would like to keep compatibility with numpty 1.x for the time being (it could be v1.26 only if needed).
I think that right now this compatibility is there no we just should not enforce `numpy>=2` in the `pytpoject.toml` file.
Moreover please consider that `ubuntu.yml` (correctly IMHO) still uses `lumpy <2`.

Reply:
A point of clarification -- this is in the *build-system* requires, not the *project dependencies*. A project built with NumPy 2.0 will have code that is both NumPy 2.0+ and Numpy 1.x compatible. So I think we're okay here?

(As some secondary proof of this beyond the [official docs](https://numpy.org/devdocs/dev/depending_on_numpy.html#numpy-2-0-specific-advice) stating this compatibility, the Ubuntu test below that installs `numpy<2` does pass without any modification.)

Reply:
The point is that I would like to be able to build PyTables from sources also on platforms where (for any reason) numpy 2 is not available. (At least for one release.)
I'm specifically thinking to Linux packaging (for e.g. Ubuntu/Debian).

In our CI I would expect that numpy 2 is used in any case event if we do not require it explicitly in the `[build-system]` section.

Reply:
... actually we're already guaranteed that the wheel build will use 2.0 simply because the `build-requirements.in`/`.txt` here pins to NumPy 2.0, so this line can just go back to what it was before #1184